### DISCUSSION TITLE
Get ID, X, Y of a device on COOJA

### DISCUSSION THREAD

**Discussion by Tagliente-Simone** [contiki-ng/contiki-ng#1759] [Q&A]
Hello everyone! Do you know how can I get the position (x and y) and the ID of a device in COOJA in the mote C script?

**Comment by atiselsts** [comment]
See here: https://stackoverflow.com/questions/64360175/get-the-position-of-node-using-cooja-contiki

**Reply by Tagliente-Simone** [reply]
Thank you, but maybe I didn't explain correctly. The idea is that using the mqtt-demo example I need to put the ID and the coordinates in the message that the node will publish in the topic. If that is not possible, the my idea is to generate those values randomly at run time.

**Reply by eric248550** [reply]
how can we use this js code to get position in cooja? cooja only take c code

**Reply by atiselsts** [reply]
@eric248550 there are two possible interpretations of the question:
1. Get position of the node from a Cooja script snippet (JavaScript code)
2. Get position of the node from a Contiki application (C code)
The original question was ambiguous. I answered for the option 1., pointing to the standard way how to do it.
For option 2. there's not standard way, although you can implement your own, by adding variables in the C code that are set from  Cooja using `setInt32ValueOf` or the similar functions.

**Reply by eric248550** [reply]
@atiselsts thanks for your answer, what I want to do is to get position of the node from a Contiki application, with the Cooja simulator. 
However, the position generated from Cooja simulator is decided in the interface, I can't find any reference or solution to implement it (although it's the common issue that been asked from years)
